packages/p2p/src/components/app.jsx

```diff
@@ -105,6 +105,12 @@ class App extends Component {
         this.setIsAdvertiser();
 
         subscribeWS({ p2p_order_list: 1, subscribe: 1 }, this.setP2pOrderList);
+
+        const url_params = new URLSearchParams(window.location.search);
+        const order_id = url_params.get('order');
+        if (order_id) {
+            this.redirectTo('orders', { order_id });
+        }
```

Comment:
@mustofa-binary we don't want to do this in the P2P application as we don't know the constraints the consumers of the library will have. Let's move this URL logic outside (to Core) and then pass the order ID in as a prop to the P2P application.

Reply:
if can, i want to avoid having too much dependency on core, can we use react-router-dom here instead?

Reply:
@mustofa-binary so far the plans for P2P is that it is to be consumed as a micro-frontend and not as a standalone application itself, in which case it's okay to be dependent on the consumer (core or binary), however, if the plans are to have it be standalone at some point, then we need to add both pieces of logic (if it is consumed then check props, otherwise if it's standalone, check URL by itself).

Reply:
that make sense, thank you 🙏 , will amend on it